chain/client/src/tests/bug_repros.rs

```diff
@@ -298,13 +298,17 @@ fn test_long_gap_between_blocks() {
                       -> (PeerManagerMessageResponse, bool) {
                     match msg.as_network_requests_ref() {
                         NetworkRequests::Approval { approval_message } => {
-                            actix::spawn(conns[1].1.send(GetBlock::latest()).then(move |res| {
-                                let res = res.unwrap().unwrap();
-                                if res.header.height > target_height {
-                                    System::current().stop();
-                                }
-                                futures::future::ready(())
-                            }));
+                            actix::spawn(
+                                conns[1].1.send(GetBlock::latest().with_span_context()).then(
+                                    move |res| {
+                                        let res = res.unwrap().unwrap();
+                                        if res.header.height > target_height {
+                                            System::current().stop();
+                                        }
+                                        futures::future::ready(())
+                                    },
+                                ),
```

Comment:
Moving the request to a variable looks good to me.
However, extracting the callback to a variable requires the argument to have an explicit type. For example, in this case it would be

```
let callback = move |res : Result<<GetBlock as actix::Message>::Result, MailboxError>| {
...
};
```

Which makes things harder to write and update. Inlining the callback is not too bad given this inconvenience.

Reply:
Alright, what if instead of outlining the closure/message, we did something like this:

```rust
let actor = conns[1].1.send(GetBlock::latest().with_span_context());
let actor = actor.then(move |res| {
    ...
});
actix::spawn(actor);
```

?

Reply:
Done.